php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -68,6 +68,10 @@ public function __construct($desc = NULL)
         // specific descriptor from the descriptor pool.
         if (get_class($this) === 'Google\Protobuf\Internal\MapEntry') {
             $this->desc = $desc;
+            foreach ($desc->getField() as $field) {
+                $setter = $field->getSetter();
+                $this->$setter($this->defaultValue($field));
```

Comment:
The `MapEntry` field has pass-by-reference setters (I don't know why, this may be a bad idea). Because of that, the default value needs to be set to a variable before being passed in, to avoid throwing a Strict Standards error. Something like this will work:

```php
$setter = $field->getSetter();
$defaultValue = $this->defaultValue($field);
$this->$setter($defaultValue);
```